src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,205 @@
+    }
+
+    @Test
+    void register_addUserPasswordIsShort_notOk() {
+        user1.setPassword(EMPTY_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(ONE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(TWO_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(THREE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(FOUR_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setPassword(FIVE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        assertEquals(0, people.size());
+    }
+
+    @Test
+    void register_addUserLoginIsShort_notOk() {
+        user1.setLogin(EMPTY_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setLogin(ONE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setLogin(TWO_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setLogin(THREE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setLogin(FOUR_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        user1.setLogin(FIVE_CHAR_STRING);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(user1);
+        });
+        assertEquals(0, people.size());
+    }
+
+    @Test
+    void register_addUserWithSameLogin_notOK() {
+        registrationService.register(user1);
+        registrationService.register(user2);
+        registrationService.register(user3);
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(new User(user1.getLogin(),"djdofjs", 18));
+        });
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(new User(user2.getLogin(),"dj10fjs", 19));
+        });
+        assertThrows(InvalidDataException.class, () -> {
+            registrationService.register(new User(user3.getLogin(),"dj20fjs", 20));
+        });
+        assertEquals(3,people.size());
+    }
+
+    @Test
+    void register_addUserWithInvalidAge_notOk() {
```

Comment:
corner case 17 must be tested
![image](https://github.com/mate-academy/jv-registration-validation/assets/20326858/c0df20c4-4c75-47a3-9902-bf05d26407a6)